### ISSUE TITLE
FPS

### ISSUE THREAD

**Issue by Rm1n90** [dbolya/yolact#357]
Hey @dbolya, 

I have trained the yolact++ with ResNet101 Backbone and `max_size=700 `. The maximum FPS that I can achieve is 12.64 on full HD video (1920x1080). 

In my project, I just need to apply the pretrained model to a part of the frame, not the whole frame so I have changed the `get_next_frame` function from 
```
    def get_next_frame(vid):
        frames = []
        for idx in range(args.video_multiframe):
            frame = vid.read()[1]
            if frame is None:
                return frames
            frames.append(frame)
        return frames
```
 to the below function to consider only part of the input video
```
    def get_next_frame(vid):
        frames = []
        for idx in range(args.video_multiframe):
            _, frame = vid.read()
            frame = frame [500:900, 500:1088]
            if frame is None:
                return frames
            frames.append(frame)
        return frames
```
But instead of getting higher FPS (since I'm feeding a part of the video), I'm getting lower FPS (11.18) than feeding the whole frame to the model. Do you have any idea why I'm getting lower FPS when feeding part of the video?

Thanks in advance!

**Comment by dbolya** [comment]
Yup, that's because you're doing the "cropping" on the CPU. You want to avoid any CPU operation like the plague because they take so incredibly long. Instead, you should do the cropping after the image has been moved to the GPU:
```py
    def transform_frame(frames):
        with torch.no_grad():
            frames = [torch.from_numpy(frame).cuda().float()[500:900, 500:1088] for frame in frames]
            return frames, transform(torch.stack(frames, 0))
```

**Comment by Rm1n90** [comment]
Hey @dbolya, Thanks a lot! It works and improved my FPS from 12.64 to almost 14 something. But, I was expecting a higher FPS since I'm feeding the cropped frame which is something like 400x400. Do you have any suggestions?

**Comment by Rm1n90** [comment]
Hey @dbolya, Thanks a lot! It works and improved my FPS from 12.64 to almost 14 something. But, I was expecting a higher FPS since I'm feeding the cropped frame which is something like 400x400. Do you have any suggestions?
Also, I have another question that how can I select different regions of the input video frame and get the mask of these regions instead of the whole input video frame? for example, I have a 1080p input video and select two regions of the input frame (one region 200x200 and another one 400x400) and then get the mask of these regions.

Thanks in advance!
